src/selectors/entities/channels.js

```diff
@@ -1209,15 +1209,25 @@ export const getDefaultChannelForTeams: (GlobalState) => RelationOneToOne<Team,
 export const getMyFirstChannelForTeams: (GlobalState) => RelationOneToOne<Team, Channel> = createSelector(
     getAllChannels,
     getMyChannelMemberships,
+    getMyTeams,
     getCurrentUser,
-    (allChannels: IDMappedObjects<Channel>, myChannelMemberships: RelationOneToOne<Channel, ChannelMembership>, currentUser: UserProfile): RelationOneToOne<Team, Channel> => {
+    (allChannels: IDMappedObjects<Channel>, myChannelMemberships: RelationOneToOne<Channel, ChannelMembership>, myTeams: Array<Team>, currentUser: UserProfile): RelationOneToOne<Team, Channel> => {
         const locale = currentUser.locale || General.DEFAULT_LOCALE;
+
         const result = {};
-        for (const channel of Object.keys(allChannels).map((key) => allChannels[key]).sort(sortChannelsByDisplayName.bind(null, locale))) {
-            if (!result[channel.team_id] && myChannelMemberships[channel.id]) {
-                result[channel.team_id] = channel;
+        for (const team of myTeams) {
+            // Get a sorted array of all channels in the team that the current user is a member of
```

Comment:
Appreciate the extra guard, but wondering if it's strictly necessary? If we added all checks downstream like this, the code could quickly become cluttered.

Reply:
It's not strictly necessary, but I don't think it really adds complexity in this case since I'm kind of just rewriting the same function in a different order by checking channel membership before sorting. There is a small functional change since we won't return channels for teams that you're not a member of, but I would've expected the function to work this way anyway.